components/kyma-environment-broker/internal/process/provisioning/lms_tenant_step.go

```diff
@@ -0,0 +1,101 @@
+package provisioning
+
+import (
+	"errors"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage"
+	"github.com/sirupsen/logrus"
+)
+
+type LmsTenantProvider interface {
+	ProvideLMSTenantID(name, region string) (string, error)
+}
+
+// provideLmsTenantStep creates (if not exists) LMS tenant and provides its ID.
+// The step does not breaks the provisioning flow.
+type provideLmsTenantStep struct {
+	tenantProvider LmsTenantProvider
+	repo           storage.Operations
+}
+
+func NewProvideLmsTenantStep(tp LmsTenantProvider, repo storage.Operations) *provideLmsTenantStep {
+	return &provideLmsTenantStep{
+		tenantProvider: tp,
+		repo:           repo,
+	}
+}
+
+func (s *provideLmsTenantStep) Name() string {
+	return "Create_LMS_Tenant"
+}
+
+func (s *provideLmsTenantStep) Run(operation internal.ProvisioningOperation, logger logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+	if operation.Lms.TenantID != "" {
+		return operation, 0, nil
+	}
+
```

Comment:
please, consider timeout for the whole step, you can use the helper method `RetryOperation` from `OperationManager`. Call to `ProvideLMSTenantID` may never work out.